DOCS-1729: Document colorblind-safe run palettes (#1600)
wandb/docs: content/en/guides/models/track/runs/run-colors.md

title: Customize run colors
---

W&B automatically assigns a color to each run that you create in your project. You can customize these colors to help you visually distinguish runs in tables and graphs. Run colors are locally scoped—on the project page, custom colors apply only to your own workspace. In reports, custom colors for runs apply only at the section level.

## Choose a color palette

W&B provides several predefined color palettes to suit different needs and preferences. You can select a palette that works best for your team and use case.

> ### How to select a color palette

Author: ```suggestion
```

1. Navigate to your W&B project
2. Click on the **Workspace** tab from the project sidebar
3. Click the **Settings** icon in the top right corner
4. Select **Runs** from the settings drawer
5. In the **Color palette** section, choose from the available options

### Available color palettes

W&B offers three color palette options:

- **Default**: The standard W&B color palette with a vibrant range of colors
- **Colorblind-safe (deuteranomaly)**: Optimized for red-green color blindness, the most common form affecting approximately 6% of males
- **Colorblind-safe (all other forms)**: Optimized for protanopia, tritanopia, and other forms of color vision deficiency

> The colorblind-safe palettes use carefully selected colors that remain distinguishable across different types of color vision deficiency while maintaining visual appeal for users with typical color vision.

Author: ```suggestion
The colorblind-safe palettes use colors that remain distinguishable across different types of color vision difference, while maintaining visual appeal for users with typical color vision.
```

## Edit individual run colors

> You can manually change the color of specific runs to customize your visualization further.

Author: ```suggestion
To manually change the color of specific runs:
```

title: Customize run colors
---

W&B automatically assigns a color to each run that you create in your project. You can customize these colors to help you visually distinguish runs in tables and graphs. Run colors are locally scoped—on the project page, custom colors apply only to your own workspace. In reports, custom colors for runs apply only at the section level.

## Choose a color palette

W&B provides several predefined color palettes to suit different needs and preferences. You can select a palette that works best for your team and use case.

> 1. Navigate to your W&B project
> 2. Click on the **Workspace** tab from the project sidebar
> 3. Click the **Settings** icon in the top right corner
> 4. Select **Runs** from the settings drawer
> 5. In the **Color palette** section, choose from the available options

Author: ```suggestion
1. Navigate to your W&B project.
2. Click on the **Workspace** tab from the project sidebar.
3. Click the **Settings** icon in the top right corner.
4. Select **Runs** from the settings drawer.
5. In the **Color palette** section, choose from the available options.
```

### Available color palettes

W&B offers three color palette options:

- **Default**: The standard W&B color palette with a vibrant range of colors

> - **Colorblind-safe (deuteranomaly)**: Optimized for red-green color blindness, the most common form affecting approximately 6% of males

Author: ```suggestion
- **Colorblind-safe (deuteranomaly)**: Optimized for red-green color blindness, the most common form.
```

> - **Colorblind-safe (all other forms)**: Optimized for protanopia, tritanopia, and other forms of color vision deficiency

Author: ```suggestion
- **Colorblind-safe (all other forms)**: Optimized for protanopia, tritanopia, and other forms of color vision difference.
```

The colorblind-safe palettes use colors that remain distinguishable across different types of color vision difference, while maintaining visual appeal for users with typical color vision.

#### Default palette

> {{< img src="/images/app_ui/run-colors-default.png" alt="W&B workspace showing runs colored with the default palette, displaying a variety of vibrant colors for different runs">}}

Author: ```suggestion
In this example plot, runs colored with the default palette, displaying a variety of vibrant colors for different runs.

{{< img src="/images/app_ui/run-colors-default.png" alt="W&B workspace showing runs colored with the default palette" >}}
```

#### Colorblind-safe (deuteranomaly) palette

> {{< img src="/images/app_ui/run-colors-colorblind-deuteranomaly.png" alt="W&B workspace showing runs colored with the colorblind-safe palette optimized for deuteranomaly (red-green color blindness), using colors that remain distinguishable for affected users">}}

Author: ```suggestion
In this example plot, runs are colored with the colorblind-safe palette optimized for deuteranomaly (red-green color blindness), using colors that remain distinguishable for affected users.

{{< img src="/images/app_ui/run-colors-colorblind-deuteranomaly.png" alt="W&B workspace showing runs colored with the colorblind-safe palette optimized for deuteranomaly (red-green color blindness)" >}}
```

#### Colorblind-safe (all other forms) palette

> {{< img src="/images/app_ui/run-colors-colorblind-other.png" alt="W&B workspace showing runs colored with the colorblind-safe palette optimized for protanopia, tritanopia, and other forms of color vision deficiency">}}

Author: ```suggestion
In this example plot, runs are colored with the colorblind-safe palette optimized for protanopia, tritanopia, and other forms of color vision difference.

{{< img src="/images/app_ui/run-colors-colorblind-other.png" alt="W&B workspace showing runs colored with the colorblind-safe palette optimized for protanopia, tritanopia, and other forms of color vision difference" >}}
```

## Choose a color palette

W&B provides several predefined color palettes to suit different needs and preferences. You can select a palette that works best for your team and use case.

1. Navigate to your W&B project.
2. Click on the **Workspace** tab from the project sidebar.
3. Click the **Settings** icon in the top right corner.
4. Select **Runs** from the settings drawer.
5. In the **Color palette** section, choose from the available options.

### Available color palettes

W&B offers three color palette options:

- **Default**: The standard W&B color palette with a vibrant range of colors
- **Colorblind-safe (deuteranomaly)**: Optimized for red-green color blindness, the most common form.
- **Colorblind-safe (all other forms)**: Optimized for protanopia, tritanopia, and other forms of color vision difference.

The colorblind-safe palettes use colors that remain distinguishable across different types of color vision difference, while maintaining visual appeal for users with typical color vision.

#### Default palette

> In this example plot, runs colored with the default palette, displaying a variety of vibrant colors for different runs.

Author: ```suggestion
In this example plot, runs are colored with the default palette, displaying a variety of vibrant colors for different runs.
```

2. Hover over the **Name** column header, click the three horizontal dots (**...**), and select **Randomize run colors** from the dropdown menu.

{{% alert %}}
The option to randomize run colors is available only after modifying the run's table in some way, such as by sorting, filtering, searching, or grouping.
{{% /alert %}}

> ## Reset run colors

Author: ```suggestion
## Color options for grouped runs

When using the **Run colors** setting, you have two options:

- **Original project colors**: Uses the default color assignment for runs
- **Key-based colors**: Colors runs based on metric or configuration values (see [Semantic run plot legends]({{< relref "color-code-runs" >}}) for details)

## Reset run colors
```

To restore the default colors for all runs in the table:

1. Click the **Runs** tab from the project sidebar.
2. Hover over the **Name** column header, click the three horizontal dots (**...**), and select **Reset colors** from the dropdown menu.

{{< img src="/images/runs/reset-run-colors.png" alt="Reset run colors in project workspace">}}

> ## Accessibility benefits
> 
> Using colorblind-safe palettes provides several important benefits:
> 
> - **Improved team collaboration**: Ensures all team members can effectively analyze and discuss experiment results, regardless of color vision abilities
> - **Better visual distinction**: Colors are selected to maximize contrast and distinguishability, even on different monitor types and lighting conditions
> - **Professional presentation**: Makes your dashboards and reports accessible for presentations and publications
> - **Inclusive design**: Demonstrates commitment to accessibility, benefiting approximately 8% of men and 0.5% of women who have some form of color vision deficiency
> 
> The selected color palette applies to all W&B visualization features, including line plots, scatter plots, parallel coordinates plots, and run tables.

Author: ```suggestion
```

> ## Color options for grouped runs

Author: This is badly placed now, needs to be near line 5.

> ## Color options for grouped runs
> 
> When using the **Run colors** setting, you have two options:
> 
> - **Original project colors**: Uses the default color assignment for runs
> - **Key-based colors**: Colors runs based on metric or configuration values (see [Semantic run plot legends]({{< relref "color-code-runs" >}}) for details)

Author: ```suggestion
```

2. Hover over the **Name** column header, click the three horizontal dots (**...**), and select **Randomize run colors** from the dropdown menu.

{{% alert %}}
The option to randomize run colors is available only after modifying the run's table in some way, such as by sorting, filtering, searching, or grouping.
{{% /alert %}}

## Color options for grouped runs

When using the **Run colors** setting, you have two options:

> - **Original project colors**: Uses the default color assignment for runs
> - **Key-based colors**: Colors runs based on metric or configuration values (see [Semantic run plot legends]({{< relref "color-code-runs" >}}) for details)

Author: ```suggestion
- **Original project colors**: Uses the default color assignment for runs.
- **Key-based colors**: Colors runs based on metric or configuration values (see [Semantic run plot legends]({{< relref "color-code-runs" >}}) for details).
```

## Reset run colors

To restore the default colors for all runs in the table:

1. Click the **Runs** tab from the project sidebar.
2. Hover over the **Name** column header, click the three horizontal dots (**...**), and select **Reset colors** from the dropdown menu.

{{< img src="/images/runs/reset-run-colors.png" alt="Reset run colors in project workspace">}}

> ## Accessibility benefits
> 
> Using colorblind-safe palettes provides several important benefits:
> 
> - **Improved team collaboration**: Ensures all team members can effectively analyze and discuss experiment results, regardless of color vision abilities
> - **Better visual distinction**: Colors are selected to maximize contrast and distinguishability, even on different monitor types and lighting conditions
> - **Professional presentation**: Makes your dashboards and reports accessible for presentations and publications
> - **Inclusive design**: Demonstrates commitment to accessibility, benefiting approximately 8% of men and 0.5% of women who have some form of color vision deficiency
> 
> The selected color palette applies to all W&B visualization features, including line plots, scatter plots, parallel coordinates plots, and run tables.

Author: ```suggestion
```
